pkg/cmd/roachtest/tests/admission_control_index_overload_tpce.go

```diff
@@ -0,0 +1,172 @@
+		nodes := c.Spec().NodeCount
+		roachNodes := c.Range(1, nodes-1)
+		workloadNode := c.Node(nodes)
+		racks := nodes - 1
+		runDuration := 4 * opts.delayBeforeIndex
+
+		t.Status("installing cockroach")
+		c.Put(ctx, t.Cockroach(), "./cockroach", roachNodes)
+
+		startOpts := option.DefaultStartOpts()
+		settings := install.MakeClusterSettings(install.NumRacksOption(racks))
+		c.Start(ctx, t.L(), startOpts, settings, roachNodes)
+
+		{
+			promCfg := &prometheus.Config{}
+			promCfg.WithPrometheusNode(workloadNode.InstallNodes()[0])
+			promCfg.WithNodeExporter(c.All().InstallNodes())
+			promCfg.WithCluster(roachNodes.InstallNodes())
+			promCfg.WithGrafanaDashboard("http://go.crdb.dev/p/snapshot-admission-control-grafana")
+			promCfg.ScrapeConfigs = append(promCfg.ScrapeConfigs, prometheus.MakeWorkloadScrapeConfig("workload",
+				"/", makeWorkloadScrapeNodes(workloadNode.InstallNodes()[0], []workloadInstance{{nodes: workloadNode}})))
+			_, cleanupFunc := setupPrometheusForRoachtest(ctx, t, c, promCfg, []workloadInstance{{nodes: workloadNode}})
+			defer cleanupFunc()
+		}
+
+		t.Status("installing docker")
+		if err := c.Install(ctx, t.L(), workloadNode, "docker"); err != nil {
+			t.Fatal(err)
+		}
+
+		importC := make(chan struct{})
+		m := c.NewMonitor(ctx, roachNodes)
+		m.Go(func(ctx context.Context) error {
+			const dockerRun = `sudo docker run cockroachdb/tpc-e:latest`
+
+			roachNodeIPs, err := c.InternalIP(ctx, t.L(), roachNodes)
+			if err != nil {
+				return err
+			}
+			roachNodeIPFlags := make([]string, len(roachNodeIPs))
+			for i, ip := range roachNodeIPs {
+				roachNodeIPFlags[i] = fmt.Sprintf("--hosts=%s", ip)
+			}
+
+			t.Status("preparing workload (~4 hours)")
+			c.Run(ctx, workloadNode, fmt.Sprintf("%s --customers=%d --racks=%d --init %s",
+				dockerRun, opts.customers, racks, roachNodeIPFlags[0]))
+			close(importC)
+
+			t.Status("running workload (~4 hours)")
+			activeCustomers := int(float64(opts.customers) * opts.activeRatio)
+			result, err := c.RunWithDetailsSingleNode(ctx, t.L(), workloadNode,
+				fmt.Sprintf("%s --customers=%d --active-customers=%d --racks=%d --duration=%s --threads=%d %s",
+					dockerRun, opts.customers, activeCustomers, racks, runDuration, opts.threads, strings.Join(roachNodeIPFlags, " ")))
+			if err != nil {
+				t.Fatal(err.Error())
+			}
+			t.L().Printf("workload output:\n%s\n", result.Stdout)
+			if strings.Contains(result.Stdout, "Reported tpsE :    --   (not between 80% and 100%)") {
+				return errors.New("invalid tpsE fraction")
+			}
+			return nil
+		})
+		m.Go(func(ctx context.Context) error {
+			// Wait for the import to complete.
+			t.Status("waiting for import to complete")
+			select {
+			case <-importC:
+			case <-ctx.Done():
+				return ctx.Err()
+			}
+
+			t.Status("waiting before creating the index")
+
+			// Wait for the load to run for a while before starting the index build.
+			select {
+			case <-time.After(opts.delayBeforeIndex):
```

Comment:
Look at how the TPC-C harness does things. It's very flexible, and better in many regards. There we have that `During` callback where test authors can inject whatever sleep they want during a test run, and this sort of thing would apply. So I would suggest improving the TPC-E test harness first as part of this PR and then adding a good use of it. It'll also encourage more TPC-E roachtests from other teams.